gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityInteractionsFetcher.java

```diff
@@ -666,8 +657,7 @@ private void parseResultSet(
                     .toArray(String[]::new));
 
         List<String> otherIdColumns =
-            getEntityIdColumnsFromInteraction(
-                DomainEntityType.valueOf(otherEntityType.toUpperCase()), incoming);
+            getEntityIdColumnsFromInteraction(otherEntityType.toUpperCase(), incoming);
```

Comment:
I'm guessing from this toUpperCase we're just hiding a conversion to enum - so the problem isn't solved. If fixing that is out of scope, then we'd have to introduce an alternative. If it's not prohibitive, maybe it makes sense to create a separate PR to be merged first that fixes this.

Reply:
I am changing this to entityType only. No uppercase, I feel like that was only a precautionary check before also.